spec/integration/import_spec.rb

```diff
@@ -0,0 +1,63 @@
+RSpec.describe 'Lazy-booting external deps' do
+  before do
+    module Test
+      class Umbrella < Dry::System::Container
+        configure do |config|
+          config.name = :core
+          config.root = SPEC_ROOT.join('fixtures/umbrella').realpath
+        end
+      end
+
+      class App < Dry::System::Container
+        configure do |config|
+          config.name = :main
+        end
+      end
+    end
+  end
+
+  shared_examples_for 'lazy booted dependency' do
+    it 'lazy boots an external dep provided by top-level container' do
+      expect(user_repo.repo).to be_instance_of(Db::Repo)
+    end
+
+    it 'loads an external dep during finalization' do
+      system.finalize!
+      expect(user_repo.repo).to be_instance_of(Db::Repo)
+    end
+  end
+
+  context 'when top-level container provides the dependency' do
```

Comment:
Is this example description mixed up with the one below? Seeing that the user repo here is importing `core.db.repo`, I thought the description would be 'when top-level container requires the dependency _from the imported container_', since `core` denotes the imported container?